gpkit/constraints/set.py

```diff
@@ -127,19 +128,17 @@ def subinplace(self, subs, value=None):
         for constraint in self:
             constraint.subinplace(subs, value)
 
-    @property
-    def varkeys(self):
-        "return all Varkeys present in this ConstraintSet"
-        return self._varkeys()
-
-    def _varkeys(self, init_dict=None):
-        "return all Varkeys present in this ConstraintSet"
-        init_dict = {} if init_dict is None else init_dict
-        out = KeySet(init_dict)
+    def reset_varkeys(self, init_dict=None):
+        "Goes through constraints and collects their varkeys."
+        varkeys = KeySet()
+        if not init_dict is None:
+            varkeys.update(init_dict)
         for constraint in self:
             if hasattr(constraint, "varkeys"):
-                out.update(constraint.varkeys)
-        return out
+                varkeys.update(constraint.varkeys)
+            # else:
+            #     print type(constraint) # getting numpy bools...
+        self.varkeys = varkeys
```

Comment:
This error occurred because there was a variable without a model, perhaps fixed by my latest commit.